### PR TITLE
EXP-2337 add getCurrent ( Device, Location, Experience, Zone), update…

### PR DESCRIPTION
**Pull request by cesar-oyarzun-m**
… readme file

### CODE REVIEW

File: Pod/Classes/Device.swift [hunk 1]
```diff
@@ -56,4 +58,32 @@ public final class Device: Model,ResponseObject,ResponseCollection {
...
+                        switch response.result{
+                        case .Success(let data):
+                            fulfill(data)
+                        case .Failure(let error):
+                            return reject(error)
+                        }
+                }
+            }
+            }else{
+                prom = Promise{fulfill,reject in}
```
Comment (adamgallowayscala) [inline Pod/Classes/Device.swift]: this promise never returns right? we should reject or throw an error if there is an error getting the identity uuid

File: Pod/Classes/Experience.swift [hunk 2]
```diff
@@ -53,4 +53,21 @@ public final class Experience: Model,ResponseObject,ResponseCollection {
...
+     Get Current Experience
+     @return Promise<Experience>.
+     */
+    public func getCurrentExperience() -> Promise<Experience>{
+        return Device.getCurrentDevice().then{ (device:Device) -> Promise<Experience> in
+            return Promise<Experience> { fulfill, reject in
+                if let experience:Experience = device.getExperience(){
+                    fulfill(experience)
+                }else{
+                    debugPrint("Experience - getCurrentExperience NULL")
```
Comment (adamgallowayscala) [inline Pod/Classes/Experience.swift]: this promise never resolves or rejects if there is no experience

Comment (cesar-oyarzun-m) [inline Pod/Classes/Experience.swift]: upps yeah I missing that I will reject the promise

**Comment by cesar-oyarzun-m** [comment]
should we ok now 👍

File: Pod/Classes/Device.swift [hunk 3]
```diff
@@ -56,4 +58,32 @@ public final class Device: Model,ResponseObject,ResponseCollection {
...
+                        switch response.result{
+                        case .Success(let data):
+                            fulfill(data)
+                        case .Failure(let error):
+                            return reject(error)
+                        }
+                }
+            }
+            }else{
+                prom = Promise(error: NSError(domain: "error", code: 100, userInfo: ["message":"Identity uuid is null"]))
```
Comment (adamgallowayscala) [inline Pod/Classes/Device.swift]: the docs say that these will resolve to null but instead they are rejecting with an error

Comment (cesar-oyarzun-m) [inline Pod/Classes/Device.swift]: yeah I will change the docs for this, I'm rejecting the promise with error message I think this is better then return null

Comment (adamgallowayscala) [inline Pod/Classes/Device.swift]: I don't think we should reject here because it's not an error, null is fine

**Comment by adamgallowayscala** [comment]
:+1:

**Pull request merged**